src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java

```diff
@@ -429,6 +429,11 @@ public ExecutableElement findMethod(TypeElement te, String methodName,
         }
     }
 
+    public boolean isRestrictedAPI(Element el) {
```

Comment:
It would be good to file a JBS issue, requesting that this info should be available using `java.lang.model` API, perhaps `util.Elements.isRestricted(ExecutableElement)`
and to document that JBS issue here.

Bottom line: ideally, the doclet should be a clean client of the Language Model API, and other public API. Workarounds are "bad" and indicative of something wrong or missing elsewhere.

Reply:
I'll file an issue. We also have similar workaround methods for preview API, I wonder if these should also be "made official"?

Reply:
I filed an issue for this, also covering preview API status: https://bugs.openjdk.org/browse/JDK-8318417